chain/indexer/src/streamer/types.rs

```diff
@@ -39,7 +39,7 @@ pub struct IndexerExecutionOutcomeWithReceipt {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct IndexerShard {
-    pub shard_id: types::ShardId,
+    pub shard_ord: types::ShardOrd,
```

Comment:
(From @bowenwang1996) @khorolets @frol will this change here break anything? Looks like the struct derives Serialize

Reply:
To answer this question I need to understand whether `shard_ord` even need to be exposed to the outside world. I feel that we should treat it as an implementation detail only useful for an efficient shard look-ups. In this case, I would only expose `shard_id`, and while we are in this migration state (we have not introduced `ShardId` yet), I suggest we keep `shard_id: ShardOrd` and later change it to `shard_id: ShardId`.

WDYT?

P.S. Maybe we want to introduce `ShardId` type now, to clearly communicate the intent here.

Reply:
@frol thanks for the detailed review! To answer your question, `shard_ord` will be exposed to the outside world, and `shard_id` is only used inside a validator's state. `shard_ord` is an ordinal number ranging from 0 to NUM_SHARDS-1 that is used on the protocol level. `shard_id` is an internal id that validators use to identify shards. The reason why they can be different is that there can be shards from different epochs after we introduce resharding. I don't think we will ever expose `shard_id`, but we should expose `shard_ord`